docs/groovy/reference/community-questions/shifts.md

```diff
@@ -0,0 +1,123 @@
+---
+title: How do row keys and positional indices behave during merge table operations?
+sidebar_label: How do row keys and positional indices behave during merge table operations?
+---
+
+_When merging tables in Deephaven, I understand that "shifts" can happen when one table grows beyond its allocated row key slots. How do the row key (k) and positional index (i) attributes change during these shifts, and what impact does this have on downstream operations?_
+
+When you merge tables in Deephaven, the engine pre-allocates row key slots for each constituent table. Understanding how row keys (`k`) and positional indices (`i`) behave during growth and shifts is crucial for working with formulas that reference these attributes.
```

Comment:
The engine also doesn't let you do it, unless you opt-in to "I am putting on my big boy pants and will get incorrect answers."  You mention that much lower down, but it is deserving of a DANGER banner.

The static example with newTable works, because it is static and these things never change.

(java.lang.IllegalArgumentException: Formula 'k' uses i, ii, k, or column array variables, and is not safe to refresh. Note that some usages, such as on an append-only table are safe. To allow unsafe refreshing formulas, set the system property io.deephaven.engine.table.impl.select.AbstractFormulaColumn.allowUnsafeRefreshingFormulas.)